### ISSUE TITLE
msmtp: account protonmail from /home/$USER/.msmtprc: cannot use tls_trust_file with tls_certcheck turned off

### ISSUE THREAD

**Issue by WhaleJ84** [LukeSmithxyz/mutt-wizard#262]
Hey, I must unfortunately open a new ticket similar to [Issue 95](https://github.com/LukeSmithxyz/mutt-wizard/issues/95) in regards to adding a Protonmail account to Mutt, as the issues raised there were in relation to the use of OfflineImap, but you've since moved to isync/msbync. 
Adding my protonmail account with `mw add` shows:

```
Do you want to keep your mail for this account offline with mbsync? [yes/no]
	yes
Insert the email address that you want to autoconfigure for mutt/mbsync
	Email: EMAIL@protonmail.com

Searching for protonmail.com in `domains.csv`...
Congrats! Server info has automatically been found, so you won't have to look anything up!
	IMAP server: 127.0.0.1
	IMAP port: 1143
	SMTP server: 127.0.0.1
	SMTP port: 1025
This data will be used by the wizard.
REMEMBER: Protonmail users must install and configure Protonmail Bridge first for the sync to work:
https://protonmail.com/bridge/
Enter the full name you want to be identified by on this account.
	Real name: FULLNAME
Enter a short, one-word identifier for this email account that will distinguish them from any other accounts you add.
	Account name: 	protonmail
If your account has a special username different from your address, insert it now. Otherwise leave this prompt totally blank.
Most accounts will not have a separate login, so you should probably leave this blank.
	Login(?): 
If you want to limit the number of messages kept offline to a number, enter that number below. If you do not want to limit your mail and would like `mbsync` to sync all mail, press enter without typing a number.
	
Enter password for mutt-wizard-protonmail: 
Retype password for mutt-wizard-protonmail: 
Creating profiles for `protonmail`...DONE.
Getting Protonmail bridge fingerprint...
msmtp: account protonmail from /home/$USER/.msmtprc: cannot use tls_trust_file with tls_certcheck turned off
sed: can't read /home/$USER/.mbsyncrc: No such file or directory
```

I am aware that the final line is due to me not having any other accounts set up yet, as for the tls_certcheck issue, I have looked through [Issue 95](https://github.com/LukeSmithxyz/mutt-wizard/issues/95) and many others, including [HOWTO: Install/Configure msmtp and mutt on ubuntu](https://wjianz.wordpress.com/2014/09/03/howto-installconfigure-msmtp-and-mutt-on-ubuntu/), which seems to be the closest to the current configuration and none of them get me closer to getting Protonmail working.

I have the Protonmail Bridge installed, logged in and can see it works okay, but when you say: 

> Protonmail accounts will require you to set up "Protonmail Bridge" to access PM's IMAP and SMTP servers. **Configure that before running mutt-wizard.**

Namely the latter bold part, I've looked around the Bridge and see nothing that needs configuring and assume by your words simply mean "get that working first" as also stated within the installer:

> REMEMBER: Protonmail users must install and configure Protonmail Bridge first for the sync to work:
> https://protonmail.com/bridge/

I've tried manually adding the information to .msmtprc but doing so makes the mutt-wizard installer fail when trying to configure the other parts.

Without knowing all the inner workings of how everything is set up, I don't know where to go to set this `tls_certcheck on` tag to allow the setup progress further.

I use Ubuntu 18.04, msmtp 1.6.6 (the newest version on the Ubuntu repos, but the current version is 1.8.5) and have Protonmail-Bridge 1.1.6, which is the most up to date version available as of 2019-07-27.

I hope this is all the information you need,
Many thanks.

**Comment by WhaleJ84** [comment]
Working on this further, I'm trying use what has been posted in Issue 95 and convert it to be used with the new setup.

Without a protonmail account successfully created, it is harder to get the SHA256 fingerprint from the Protonmail certificate, but I've narrowed that down to this command:

`msmtp --serverinfo --host=127.0.0.1 --port=1025 --tls=on --tls-certcheck=off`

**Comment by WhaleJ84** [comment]
If I edit the fingerprint function online 197 in /usr/bin/mw to:

`--tls-certcheck=on`

I get the error:

`msmtp: TLS certificate verification failed: the certificate hasn't got a known issuer`

and adding /home/$USER/.config/protonmail/bridge/cert.pem to the certs in line 29 also gives the same result.

Now I need to try and get the certificate known or locate a known one on my system if there is one.
Any tips?

**Comment by WhaleJ84** [comment]
So I think I've fixed the issue and will list my commands below in hopes this will help others.

With msmtp complaining the certificate isn't legit I manually added protonmail myself into ca-certificates.crt:

```
cp /etc/ssl/certs/ca-certificates.crt ~/Documents
cp /home/$USER/.config/protonmail/bridge/cert.pem ~/Documents
cat cert.pem >> ca-certificates.crt
fgrep -xf cert.pem ca-certificates.crt
sudo mv ca-certificates.crt /usr/local/share/ca-certificates/
sudo update-ca-certificates
```

I then had to edit line 29 in /usr/bin/mw to include /usr/local/share/ca-certificates.crt to the list of certs mutt-wizard looks for.

Line 197 still does my edit of `msmtp --serverinfo --host=127.0.0.1 --port=1025 --tls=on --tls-certcheck=off` as your previous implementation (as far as I could tell) looks for a successfully created account called protonmail - something that isn't possible, whereas this version _should_ work for all Protonmail users.

I can successfully download my mail and send new ones with these changes and would like to see if it works for others too.
